src/notation/view/selectionfiltermodel.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "selectionfiltermodel.h"
+
+#include "engraving/libmscore/select.h"
+
+#include "translation.h"
+
+using namespace mu::notation;
+
+SelectionFilterModel::SelectionFilterModel(QObject* parent)
+    : QAbstractListModel(parent)
+{
+}
+
+void SelectionFilterModel::load()
+{
+    beginResetModel();
+
+    m_types.clear();
+    m_types << SelectionFilterType::ALL;
+
+    for (int i = 0; i < Ms::NUMBER_OF_SELECTION_FILTER_TYPES; i++) {
+        m_types << static_cast<SelectionFilterType>(1 << i);
+    }
+
+    updateIsAllSelected();
+    endResetModel();
+
+    globalContext()->currentNotationChanged().onNotify(this, [this]() {
+        emit enabledChanged();
+
+        if (currentNotation()) {
+            updateIsAllSelected();
+            emit dataChanged(index(1), index(rowCount() - 1), { IsSelectedRole });
+        }
+    });
+}
+
+void SelectionFilterModel::updateIsAllSelected()
+{
+    if (!currentNotationInteraction()) {
+        return;
+    }
+
+    m_isNoneSelected = true;
+    m_isAllSelected = true;
+
+    // Skip "All" type
+    for (auto type : m_types.mid(1)) {
```

Comment:
I think, better to add more explicit skipping for this element:

for (SelectionFilterType type : m_types) {
if (type == All) {
continue;
}
...
}

Because this code will break if we change the position of the All type in the list